tests/monitoring_processes/test_monitoring_critical_processes.py

```diff
@@ -0,0 +1,401 @@
+        container_name: A string shows container name.
+        program_name: A string shows process name.
+        program_pid: An integer represents the PID of a process.
+
+    Returns:
+        None.
+    """
+    if program_status == "RUNNING":
+        kill_process_by_pid(duthost, container_name, program_name, program_pid)
+    elif program_status in ["EXITED", "STOPPED", "STARTING"]:
+        pytest.fail("Program '{}' in container '{}' is in the '{}' state, expected 'RUNNING'"
+                    .format(program_name, container_name, program_status))
+    else:
+        pytest.fail("Failed to find program '{}' in container '{}'"
+                    .format(program_name, container_name))
+
+
+def stop_critical_processes(duthost, containers_in_namespaces):
+    """Gets critical processes of each running container and then stops them from running.
+
+    Args:
+        duthost: Hostname of DUT.
+        containers_in_namespaces: A dictionary where keys are container names and
+        values are lists which contains ids of namespaces this container should reside in.
+
+    Returns:
+        None.
+    """
+    for container_name in containers_in_namespaces.keys():
+        critical_group_list, critical_process_list, succeeded = duthost.get_critical_group_and_process_lists(container_name)
+        pytest_assert(succeeded, "Failed to get critical group and process lists of container '{}'".format(container_name))
+
+        namespace_ids = containers_in_namespaces[container_name]
+        for namespace_id in namespace_ids:
+            container_name_in_namespace = container_name
+            if namespace_id != DEFAULT_ASIC_ID:
+                container_name_in_namespace += namespace_id
+
+            for critical_process in critical_process_list:
+                # Skip 'dsserve' process since it was not managed by supervisord
+                # TODO: Should remove the following two lines once the issue was solved in the image.
+                if container_name_in_namespace == "syncd" and critical_process == "dsserve":
+                    continue
+
+                program_status, program_pid = get_program_info(duthost, container_name_in_namespace, critical_process)
+                check_and_kill_process(duthost, container_name_in_namespace, critical_process, program_status, program_pid)
+
+            for critical_group in critical_group_list:
+                group_program_info = get_group_program_info(duthost, container_name, critical_group)
+                for program_name in group_program_info:
+                    check_and_kill_process(duthost, container_name_in_namespace, program_name,
+                                           group_program_info[program_name][0],
+                                           group_program_info[program_name][1])
+
+
+def check_and_restart_process(duthost, container_name, critical_process):
+    """Checks the running status of a critical process and restarts it if it was not running.
```

Comment:
If the process is stopped, we don't need to restart it. We should simply start it.